### PR TITLE
Always use LockService from Factory when running command steps

### PR DESCRIPTION
**Pull request by filipelautert**
## Impact

- [X] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description

We should not store LockService as database listeners may change. Always ask it from the factory. This is causing errors when running the refactored UpdateSql command. as per issue https://github.com/liquibase/liquibase/issues/4150

### CODE REVIEW

**Review by abrackx — APPROVED** [review]
Is it possible to get a test set up for this? Change looks good either way. Thanks!

**Review by StevenMassaro — APPROVED** [review]

**Review by suryaaki2 — APPROVED** [review]

**Comment by XDelphiGrl** [comment]
Functional Tests Started here: [https://github.com/liquibase/liquibase-pro-tests/actions/runs/4851822443](https://github.com/liquibase/liquibase-pro-tests/actions/runs/4851822443)

Test Harness execution started here: [https://github.com/liquibase/liquibase-test-harness/actions/runs/4851828531](https://github.com/liquibase/liquibase-test-harness/actions/runs/4851828531)

File: liquibase-standard/src/test/groovy/liquibase/command/UpdateCommandsTest.groovy [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+        connection.createStatement().execute("DROP ALL OBJECTS")
+
+        database = new H2Database()
+        database.setConnection(new JdbcConnection(connection))
+    }
+
+    def "run UpdateSql from CommandStep"() {
+        when:
+        def updateCommand = new CommandScope(UpdateSqlCommandStep.COMMAND_NAME)
```
Comment (XDelphiGrl) [inline liquibase-standard/src/test/groovy/liquibase/command/UpdateCommandsTest.groovy:37]: The `updateCommand`  variable name for this case should be `updateSqlCommand` to help with readability of the test.

**Review by XDelphiGrl — APPROVED** [review]
Liquibase recently refactored many Liquibase command classes, including UpdateSqlCommand. This  introduced a bug in UpdateSqlCommand whereby calling UpdateSqlCommand from Java threw an exception trying to unlock the non-existent DATABASECHANGELOGLOCK table. This PR addresses the bug for users calling UpdateSqlCommand from Java.
* This bug did not impact the CLI.
*  A test build of the fix was validated by @dominiquems in [https://github.com/liquibase/liquibase/issues/4150#issuecomment-1513864711](https://github.com/liquibase/liquibase/issues/4150#issuecomment-1513864711).
* New integration test added for `UpdateSqlCommand`.
* Functional and test harness executions passing.

APPROVED

**Comment by sideeffffect** [comment]
Does this PR also fix the problem of logs going straight to `stdout` (instead of j.u.l)?

**Pull request merged**